plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -140,20 +146,48 @@ public ConnectorTableMetadata getTableMetadata()
         return connectorTableMetadata;
     }
 
-    private Optional<RowType> getPartitionColumnType(List<PartitionField> fields, Schema schema)
+    private static List<PartitionField> getAllPartitionFields(Table icebergTable)
     {
+        Set<Integer> existingColumnsIds = icebergTable.schema()
+                .columns().stream()
+                .map(NestedField::fieldId)
+                .collect(toUnmodifiableSet());
+
+        List<PartitionField> visiblePartitionFields = icebergTable.specs()
+                .values().stream()
+                .flatMap(partitionSpec -> partitionSpec.fields().stream())
+                // skip columns that were dropped
+                .filter(partitionField -> existingColumnsIds.contains(partitionField.sourceId()))
+                .collect(toImmutableList());
+
+        Set<Integer> alreadyExistingFieldIds = new HashSet<>();
```

Comment:
nit (optional): extract the logic for filtering duplicates to a different method. (just to improve the readability)